perf/benchmark/security/generate_policies/generate_policies.go

```diff
@@ -0,0 +1,188 @@
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+
+	"sort"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/protobuf/proto"
+
+	"istio.io/istio/pkg/util/protomarshal"
+
+	authzpb "istio.io/api/security/v1beta1"
+)
+
+type ruleOption struct {
+	occurrence int
+	gen        generator
+}
+
+type MyPolicy struct {
+	APIVersion string         `json:"apiVersion"`
+	Kind       string         `json:"kind"`
+	Metadata   MetadataStruct `json:"metadata"`
+}
+
+type MetadataStruct struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+func ToYAML(policy *MyPolicy, spec proto.Message) (string, error) {
+	header, err := json.Marshal(policy)
+	if err != nil {
+		return "", err
+	}
+
+	headerYaml, err := yaml.JSONToYAML(header)
+	if err != nil {
+		return "", err
+	}
+
+	authorizationPolicy, err := protomarshal.ToYAML(spec)
+	if err != nil {
+		return "", err
+	}
+
+	rulesYaml := bytes.Buffer{}
+	rulesYaml.WriteString("spec:\n")
+	scanner := bufio.NewScanner(strings.NewReader(authorizationPolicy))
+	for scanner.Scan() {
+		rulesYaml.WriteString(" " + scanner.Text() + "\n")
+	}
+	return string(headerYaml) + rulesYaml.String(), nil
+}
+
+func getOrderedKeySlice(ruleToOccurrences map[string]*ruleOption) *[]string {
+	var sortedKeys []string
+	for key := range ruleToOccurrences {
+		sortedKeys = append(sortedKeys, key)
+	}
+	sort.Strings(sortedKeys)
+	return &sortedKeys
+}
+
+func generateAuthorizationPolicy(action string, ruleToOccurrences map[string]*ruleOption, policy *MyPolicy) (string, error) {
+	spec := &authzpb.AuthorizationPolicy{}
+	switch action {
+	case "ALLOW":
+		spec.Action = authzpb.AuthorizationPolicy_ALLOW
+	case "DENY":
+		spec.Action = authzpb.AuthorizationPolicy_DENY
+	}
+
+	var ruleList []*authzpb.Rule
+	sortedKeys := getOrderedKeySlice(ruleToOccurrences)
+	for _, name := range *sortedKeys {
+		ruleOp := ruleToOccurrences[name]
+		rule := ruleOp.gen.generate(name, ruleOp.occurrence, action)
+		ruleList = append(ruleList, rule)
+	}
+	spec.Rules = ruleList
+
+	yaml, err := ToYAML(policy, spec)
+	if err != nil {
+		return "", err
+	}
+	return yaml, nil
+}
+
+func generateRule(action string, ruleToOccurrences map[string]*ruleOption,
+	policy *MyPolicy) (string, error) {
+
+	switch policy.Kind {
+	case "AuthorizationPolicy":
+		return generateAuthorizationPolicy(action, ruleToOccurrences, policy)
+	case "PeerAuthentication":
+		return "", fmt.Errorf("unimplemented")
+	case "RequestAuthentication":
+		return "", fmt.Errorf("unimplemented")
+	default:
+		return "", fmt.Errorf("invalid policy")
+	}
+}
+
+func createRules(action string, ruleToOccurrences map[string]*ruleOption, policy *MyPolicy) (string, error) {
+	yaml, err := generateRule(action, ruleToOccurrences, policy)
+	if err != nil {
+		return "", err
+	}
+	return yaml, nil
+}
+
+func createPolicyHeader(namespace string, name string, kind string) *MyPolicy {
+	metadata := MetadataStruct{namespace, name}
+	return &MyPolicy{
```

Comment:
Just and use the explicit field name in the struct so that it's easier for refactor later:
```go
return &MyPolicy{
		APIVersion: "security.istio.io/v1beta1",
		Kind:       kind,
		Metadata:   MetadataStruct{Namespace: namespace, Name: name},
	}
```

Reply:
true, updated